### PR TITLE
Usability improvement mxnet vae

### PR DESCRIPTION
**Pull request by anirudh2290**
## Description ##
Usability improvement for mxnet vae

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Added args to support running the script on CPU
- [x] Supports python3
- [x] Script can now run with all defaults
- [x] Creates directories which don't exist
- [x] Downloads default dataset if no other dataset provided by user.
- [x] Doc strings, and other small code changes

@shujonnaha

### CODE REVIEW

File: example/mxnet_adversarial_vae/README.md [hunk 1]
```diff
@@ -1,22 +1,55 @@
-VAE-GAN in MXNet
+# VAE-GAN in MXNet
 
-Implementation of Autoencoding beyond pixels using a learned similarity metric based on the Tensorflow implementation of https://github.com/JeremyCCHsu/tf-vaegan/
+* Implementation of Autoencoding beyond pixels using a learned similarity metric based on the Tensorflow implementation of https://github.com/JeremyCCHsu/tf-vaegan/
 
-*Please refer to their official Github for details*: https://github.com/andersbll/autoencoding_beyond_pixels
+* Please refer to their official Github for details*: https://github.com/andersbll/autoencoding_beyond_pixels
```
Comment (madjam) [inline example/mxnet_adversarial_vae/README.md]: details* -> details

File: example/mxnet_adversarial_vae/README.md [hunk 2]
```diff
@@ -1,22 +1,55 @@
...
-Usage
+# Prerequisites
 
-Using existing models
+* Opencv
+* Python packages required: scipy, scikit-learn and Pillow, opencv python package
 
+# Environment Tested On
+
+Ubuntu Deep Learning AMI, p2.8xlarge
```
Comment (madjam) [inline example/mxnet_adversarial_vae/README.md]: p2.8xlarge running Amazon EC2 Ubuntu Deep Learning AMI // Can you include the AMI version?

File: example/mxnet_adversarial_vae/README.md [hunk 3]
```diff
@@ -1,22 +1,55 @@
-VAE-GAN in MXNet
+# VAE-GAN in MXNet
 
-Implementation of Autoencoding beyond pixels using a learned similarity metric based on the Tensorflow implementation of https://github.com/JeremyCCHsu/tf-vaegan/
```
Comment (madjam) [inline example/mxnet_adversarial_vae/README.md]: Can you hyperlink "Autoencoding beyond pixels using a learned similarity metric" to the paper: https://arxiv.org/abs/1512.09300

Comment (madjam) [reply example/mxnet_adversarial_vae/README.md]: New sentence for: "Based on the Tensorflow implementation ..."

File: example/mxnet_adversarial_vae/README.md (hunk 1 above)
Comment (madjam) [inline example/mxnet_adversarial_vae/README.md]: details* -> details

File: example/mxnet_adversarial_vae/README.md (hunk 2 above)
Comment (madjam) [inline example/mxnet_adversarial_vae/README.md]: Amazon EC2 p2.8xlarge running Ubuntu DeepLearning AMI // can you add version?

**Comment by anirudh2290** [comment]
@madjam I have made the required changes!

**Pull request merged**